### PR TITLE
Add choose visit date and time slot

### PR DESCRIPTION
**Pull request by psoleckimoj**

### CODE REVIEW

File: server/@types/bapv.d.ts [hunk 1]
```diff
@@ -143,3 +143,23 @@ export type Contact = {
   addresses: Address[]
   commentText: string
 }
+
+// Visit slots, for representing data derived from VisitSessions
```
Comment (tpmcgowan) [inline server/@types/bapv.d.ts:147]: Happy to discuss naming of these - 'slots' vs 'sessions', etc! (Might try and order this file better too once this PRs in...)

File: server/services/visitSessionsService.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        slotsForDay.slots.morning.push(newSlot)
+      } else {
+        slotsForDay.slots.afternoon.push(newSlot)
+      }
+      return slotList
+    }, {})
+  }
+
+  // include minutes only if not on the hour; e.g. 10am / 10:30am
+  private getFormattedTime(time: Date): string {
```
Comment (tpmcgowan) [inline server/services/visitSessionsService.ts:55]: This works OK for now, but thinking it probably makes sense to move this to utils or add as a Nunjucks filter (and hence just pass timestamps to template).

File: server/@types/bapv.d.ts (hunk 1 above)
Comment (psoleckimoj) [reply server/@types/bapv.d.ts:147]: Can go with whatever we use, one for the team 🙂

**Review by tpmcgowan — APPROVED** [review]
👍

**Pull request merged**